### PR TITLE
make buildable in dotnet

### PR DESCRIPTION
**Pull request by the-daniel-rothig**
Previously you needed msbuild to build + run the UI. No more! This adds the capability to run this with https in kestrel (dotnet core) using a self-signed certificate, and adds a how-to to the readme

also includes some structural changes to the read me to split out the Visual Studio only information from general know-how

### CODE REVIEW

File: src/ui/Program.cs [hunk 1]
```diff
@@ -10,10 +11,28 @@ public static void Main(string[] args)
...
-                .UseStartup<Startup>()
-                .Build();
+        public static IWebHost BuildWebHost(string[] args) {
+            if (System.Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "Development") 
+            {
+                return WebHost.CreateDefaultBuilder(args)
+                    .UseKestrel(options =>
+                    {
+                        options.AddServerHeader = false;
+                        options.Listen(IPAddress.Loopback, 44364, listenOptions =>
```
Comment (defong) [inline src/ui/Program.cs]: 44364 now breaks 12 factor

Comment (defong) [reply src/ui/Program.cs]: consider config file usage

**Comment by dankmitchell** [comment]
This branch no longer builds for me @the-daniel-rothig 😢 .

A random port id being set each time I run `dotnet run` so auth redirect fails.

```
$ dotnet run
/usr/local/share/dotnet/sdk/2.1.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.ObsoleteReferences.targets(33,5): warning : The tool 'Microsoft.Extensions.SecretManager.Tools' is now included in the .NET Core SDK. Information on resolving this warning is available at (https://aka.ms/dotnetclitools-in-box). [/Users/Dan/Git/manage-courses-ui/src/ui/ManageCoursesUi.csproj]
/usr/local/share/dotnet/sdk/2.1.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.ObsoleteReferences.targets(33,5): warning : The tool 'Microsoft.Extensions.SecretManager.Tools' is now included in the .NET Core SDK. Information on resolving this warning is available at (https://aka.ms/dotnetclitools-in-box). [/Users/Dan/Git/manage-courses-ui/src/ui/ManageCoursesUi.csproj]
Using launch settings from /Users/Dan/Git/manage-courses-ui/src/ui/Properties/launchSettings.json...
info: GovUk.Education.ManageCourses.Ui.Startup[0]
      Using API base URL:
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using '/Users/Dan/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Overriding address(es) 'https://localhost:44364/'. Binding to endpoints defined in UseKestrel() instead.
Hosting environment: Development
Content root path: /Users/Dan/Git/manage-courses-ui/src/ui
Now listening on: https://127.0.0.1:50125
Application started. Press Ctrl+C to shut down.
```

File: src/ui/Program.cs [hunk 2]
```diff
@@ -1,19 +1,57 @@
...
+        }
+
+        public static IWebHost BuildWebHost (string[] args) {
+
+            return WebHost.CreateDefaultBuilder (args)
+                .UseKestrel (options => {
+                    options.AddServerHeader = false;
+                    var environment = System.Environment.GetEnvironmentVariable ("ASPNETCORE_ENVIRONMENT");
+                    if (environment == "Development") {
+                        var port = GetPort (environment);
```
Comment (defong) [inline src/ui/Program.cs]: @the-daniel-rothig 

I have refactored the code to be more hygienic especially with the ports as not setting or setting invalid values causes random port assignments. 

I have also expand/amended the readme as well

**Pull request merged**